### ISSUE TITLE
Combined DPS Not Displayed for Toxic Rain

### ISSUE THREAD

**Issue by masonk** [PathOfBuildingCommunity/PathOfBuilding#1293]
v 1.4.170.7 (latest)

At some point, the Combined DPS line for Toxic Rain stopped displaying in my POB.

My POB that has it broken: https://pastebin.com/rBMs8nPb
A similar character where the line is displayed: https://pastebin.com/rNigbdkq

If I import my character's gear (elmasone / tidepodavenger) into the second (working) POB, the TR Combined DPS number stops showing. This makes me suspect that some piece of my gear is bugging out POB.

**Comment by masonk** [comment]
Did some more digging. I am now certain this is a POB bug.

The guy whose POB works has this jewel:

```
Foul Lens
Searching Eye Jewel
Unique ID: 61f39806aa7bc3daba0e1965e18c3e6938734914b6754b11901728d7db7c4125
Item Level: 81
LevelReq: 62
Implicits: 1
Regenerate 0.1% of Life per second
Adds 5 to 6 Chaos Damage to Attacks
+216 to Accuracy Rating
+30 to maximum Life
Adds 8 to 14 Fire Damage to Bow Attacks
```

If you take that jewel off of his build, the combo dps number drops out of his sidebar.

If you add that jewel to my broken POB, the combo DPS number pops in.

There is something about this jewel that is related to the bug.

**Comment by masonk** [comment]
That's what it is.

His ghastly has fire damage. It adds ignite to his DPS. That's why his gets combo damage.

If I give myself any fire damage and get some ignite damage, combo DPS pops back into the sidebar.

I think the fix to this bug is to always show combo DPS if the skill has a poison damage component.

**Comment by Regisle** [comment]
Similar issue to #1108 see comment https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues/1108#issuecomment-651406969 or read entire thread for more info

**Comment by masonk** [comment]
I read the thread. I don't think this should considered WAI.

In some builds there is a large difference between DOT and Combined DPS. I posted in that original thread, thanks for the reference.

**Comment by masonk** [comment]
I'll leave this open until we see what the resolution is on the original bug.

**Comment by Quotae** [comment]
Turns out this actually was sort of a mistake. Basically, the Combined DPS output shows up whenever the internal Combined DPS value is not equal to the "Total DPS Inc. Poison" output(or Ignite/Bleed/Impale). Due to an oversight, Combined DPS and Total DPS Inc. Poison were *always* the same value if you didn't also have Ignite/Bleed/Impale damage. Total DPS Inc. Poison doesn't show up if you're using a DoT skill, so those two things combined meant nothing would ever show up.

Made a PR, #1296 , which will fix this. It makes it so Combined DPS will show up if the skill has DoT DPS as well as some Poison/Ignite/Bleed/Impale DPS.

**Comment by masonk** [comment]
Quotae, that's awesome follow up! Thanks for doing it!

One question for you: why did you decide to fix it this way instead of just making TotalDPSIncPoison show up whenever there is a poison component?

**Comment by Quotae** [comment]
Because Combined DPS is for 3 sources of DPS or more. In this case it's got Hit DPS, DoT DPS, and Poison DPS. "Total DPS Inc. Poison" is for when there is just Hit DPS and Poison DPS. Anywho, I also just added "Total DPS Inc. DoT" which is just hit + DoT DPS when you don't have any Poison/Ignite/etc DPS.

I got a couple more opinions and I also added a "Total DoT DPS" output for DoT skills that doesn't include hit DPS, just so you can see how much of your damage is actually done over time. There's cases where you might care about your hit DPS, but in most cases basically all of your DPS with DoT skills will come from the DoTs. This output shows up when you have DoT DPS + Poison/Ignite/etc.

**Comment by ppoelzl** [comment]
Fixed in 1.4.170.12.
